.changelog/3602.txt

```diff
@@ -0,0 +1,4 @@
+```release-note:bug
+cli: Update deployment destroy to proceed with deleting resources for failed
+deployments
```

Comment:
```suggestion
cli: `deployment destroy` will now attempt to destroy any known resources for failed
deployments. Previously, `destroy` would skip unsuccessful deployments.
```

I think it's important to convey that we will *attempt* to destroy the resources. Since it failed to deploy in the first place, we may not be successful, or even have all the resources tracked